core/services/workflows/engine.go

```diff
@@ -791,6 +796,7 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 		logCustMsg(ctx, cma, lmsg, l)
 		stepStatus = store.StatusCompleted
 	}
+	e.metrics.with(platform.KeyStepRef, msg.stepRef, "status", stepStatus).incrementWorkflowStepFinishedCounter(ctx)
```

Comment:
Any metrics w/ stepRef or workflowExecutionID will be high cardinality

Reply:
stepRef should not be an issue here.

Reply:
For future context: stepRef is static, has no UUID-esque values, and is not expected to change over the lifecycle of a workflow ID.